mlrun/runtimes/pod.py

```diff
@@ -1057,6 +1057,12 @@ def is_env_exists(self, name):
                 return True
         return False
 
+    def enrich_runtime_spec(
+        self,
+        project_node_selector: dict[str, str],
+    ):
+        self.spec.node_selector = {**project_node_selector, **self.spec.node_selector}
+
```

Comment:
This merge logic is very ambiguous, it might be clear to someone who knows the `**` notation very well and how it decides orders of precedence. But it's worth leaving here a very detailed description on how this merge behaves and what it does.
If this merge exists elsewhere in the same ambiguous fashion, lets create a mlrun helper function `merge_dicts` with the detailed description there